pandas/core/indexes/base.py

```diff
@@ -4636,8 +4636,9 @@ def join(
             return self._join_non_unique(other, how=how)
         elif not self.is_unique or not other.is_unique:
             if self.is_monotonic_increasing and other.is_monotonic_increasing:
-                if self._can_use_libjoin:
+                if not is_interval_dtype(self.dtype):
                     # otherwise we will fall through to _join_via_get_indexer
+                    # go through object dtype for ea till engine is supported properly
```

Comment:
Do we have a GH issue related to an engine for EAs?

Reply:
#39133